### PR TITLE
ICT spec updates and WFB payload fix

### PR DESCRIPTION
**Pull request by JesseMckinzie**
- Update ict spec to not require the UI component
- Update pydantic class to only validate UI when present
- Add function to add missing inputs in WFB payload using the links and ICT data
- Add unit tests to for add missing WFB inputs

### CODE REVIEW

**Review by sameeul — APPROVED** [review]

File: src/sophios/api/utils/converter.py [hunk 1]
```diff
@@ -161,3 +161,56 @@ def ict_to_clt(ict: Union[ICT, Path, str, dict], network_access: bool = False) -
     ict_local = ict if isinstance(ict, ICT) else cast_to_ict(ict)
 
     return ict_local.to_clt(network_access=network_access)
+
+
+def update_payload_missing_inputs_outputs(wfb_data: dict) -> dict:
```
Comment (vjaganat90) [inline src/sophios/api/utils/converter.py:166]: use more restrictive Json type instead of dict

File: src/sophios/api/utils/ict/ict_spec/cast.py [hunk 2]
```diff
@@ -22,6 +23,10 @@ def cast_to_ict(ict: Union[Path, str, dict]) -> ICT:
         else:
             raise ValueError(f"File extension not supported: {ict}")
 
+        data.pop("ui", None)
```
Comment (vjaganat90) [inline src/sophios/api/utils/ict/ict_spec/cast.py:26]: This part of the code is not tested

File: src/sophios/api/utils/ict/ict_spec/model.py [hunk 3]
```diff
@@ -24,29 +24,31 @@ class ICT(Metadata):
...
+            io_dict = {"inputs": [], "outputs": []}  # type: ignore
+            ui_keys = [ui.key.root.split(".") for ui in self.ui]
+            for ui_ in ui_keys:
+                io_dict[ui_[0]].append(ui_[1])
+            input_names = [io.name for io in self.inputs]
+            output_names = [io.name for io in self.outputs]
+            inp_bool = [x in input_names for x in io_dict["inputs"]]
+            out_bool = [x in output_names for x in io_dict["outputs"]]
+
+            if not all(inp_bool):
```
Comment (vjaganat90) [inline src/sophios/api/utils/ict/ict_spec/model.py:43]: is this part tested with converting any of the plugins to clt?

**Review by vjaganat90 — CHANGES_REQUESTED** [review]
We need to make sure this works with conversion to clt.

File: src/sophios/api/utils/converter.py [hunk 4]
```diff
@@ -161,3 +161,56 @@ def ict_to_clt(ict: Union[ICT, Path, str, dict], network_access: bool = False) -
...
+        missing_input_key = input_directories[link["inletIndex"]]["name"]
+        missing_output_key = output_directories[link["outletIndex"]]["name"]
+
+        # add the missing input value to the node if needed
+        target_node["settings"]["inputs"][missing_input_key] = source_node["settings"]["outputs"][missing_output_key]
+
+    return wfb_data_copy
+
+
+def is_inlet(binding: dict) -> bool:
```
Comment (vjaganat90) [inline src/sophios/api/utils/converter.py:208]: This function definition should be inside update_payload_missing_inputs_outputs. Shouldn't be in the outer namespace of the file.

**Pull request merged**